presto-main/src/main/java/com/facebook/presto/sql/planner/plan/MarkDistinctNode.java

```diff
@@ -41,12 +43,14 @@
     public MarkDistinctNode(@JsonProperty("id") PlanNodeId id,
             @JsonProperty("source") PlanNode source,
             @JsonProperty("markerSymbol") Symbol markerSymbol,
+            @JsonProperty("markerVariable") VariableReferenceExpression markerVariable,
             @JsonProperty("distinctSymbols") List<Symbol> distinctSymbols,
```

Comment:
distinctSymbols/hashSymbol should be VariableReference?

Reply:
One thing at a time. Otherwise each diff will be huge.